### PR TITLE
Add value to owned items if filter is true and fix tests.

### PR DESCRIPTION
**Pull request by Ecmelt**

### CODE REVIEW

File: src/cheapestTree.ts [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
 import { NestedRecipe } from '@gw2efficiency/recipe-nesting'
 
 export function cheapestTree(
   amount: number,
   tree: NestedRecipe,
   itemPrices: Record<string, number>,
   availableItems: Record<string, number> = {},
-  forceBuyItems: Array<number> = []
+  forceBuyItems: Array<number> = [],
+  valueOwnItems: string = 'false'
```
Comment (queicherius) [inline src/cheapestTree.ts:13]: I think this makes sense to be `false` (not the string). Let's just use the user's material settings, `BUY | SELL` is overkill for this (we can add it later if requested)

File: src/cheapestTree.ts [hunk 2]
```diff
@@ -32,3 +46,34 @@ export function cheapestTree(
...
 }
+
+function getCheaperToBuyItemIds(
+  tree: RecipeTreeWithPrices,
+  ids: Array<number> = []
+): Array<number> {
+  if (
+    typeof tree.craftDecisionPrice === 'number' &&
+    typeof tree.buyPrice === 'number' &&
+    tree.craftDecisionPrice * 0.85 > tree.buyPrice
```
Comment (queicherius) [inline src/cheapestTree.ts:57]: I think this is easier to understand flipped and with an explainer comment:

```ts
// If buying the item is less gold than we can get by selling the ingredients (incl. TP tax), we should just buy the item instead
tree.buyPrice < tree.craftDecisionPrice * 0.85
```

(Feel free to reword)

File: src/cheapestTree.ts [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
+      amount,
+      tree as RecipeTree,
+      {}
+    )
+    const treeWithPriceWithoutAvailableItems = calculateTreePrices(
+      treeWithQuantityWithoutAvailableItems,
+      itemPrices
+    )
+    const cheaperToBuyItemIds = getCheaperToBuyItemIds(treeWithPriceWithoutAvailableItems)
+    disableCraftForItemIds(tree, cheaperToBuyItemIds)
```
Comment (queicherius) [inline src/cheapestTree.ts:26]: Does this not have to be below L33?

File: tests/__snapshots__/calculateTreeCraftFlags.spec.ts.snap [hunk 4]
```diff
@@ -17,6 +17,7 @@ Object {
       "disciplines": Array [],
       "id": 2,
       "min_rating": null,
+      "multipleRecipeCount": 1,
```
Comment (queicherius) [inline tests/__snapshots__/calculateTreeCraftFlags.spec.ts.snap:20]: This would be easier to review in another PR with just this change.

**Review by queicherius — CHANGES_REQUESTED** [review]
Looks mostly good, please check the comments and add a test for this change

File: src/cheapestTree.ts (hunk 3 above)
Comment (queicherius) [reply src/cheapestTree.ts:27]: Ah, that makes sense. Could you add a quick oneliner comment above that ("calculateTreeQuantity already checks for craft flags, so we can adjust it here")

File: package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@gw2efficiency/recipe-calculation",
   "description": "Calculate the cheapest tree traversal, price and used items of crafting recipes",
-  "version": "3.2.1",
+  "version": "3.3.0",
```
Comment (queicherius) [inline package.json:4]: Could you undo this? Else it likely breaks our release tooling

**Review by queicherius — CHANGES_REQUESTED** [review]

**Comment by Ecmelt** [comment]
@queicherius the test fails with `Argument of type 'true' is not assignable to parameter of type 'false'.`

**Review by queicherius — APPROVED** [review]

**Pull request merged**